src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -560,7 +560,9 @@ export function registerTerminalActions() {
 				text = editor.getModel().getValueInRange(selection, endOfLinePreference);
 			}
 			instance.sendText(text, true);
-			return terminalGroupService.showPanel();
+			if (terminalService.configHelper.config.showPanelOnRunSelectedText) {
```

Comment:
thanks for the PR. we want to avoid adding settings for cases like this. 

What would be better here is to check if  `(instance.target === TerminalLocation.Panel)`